clients/drcachesim/tests/memref_gen.h

```diff
@@ -137,6 +145,42 @@ gen_exit(memref_tid_t tid)
     return memref;
 }
 
-} // namespace
+/* Returns a vector of memref_t with instr encodings.
+ * The caller has to set tid + pid fields of the memref_t in memref_instr_t
+ * structs but not the other fields. Also note that all data memrefs have to
+ * be filled in for each instr when constructing memref_instr_vec. This only
+ * applies to instr_t and for other memrefs, the caller should set everything
+ * they need.
+ */
+inline std::vector<memref_t>
+get_memrefs_from_ir(instrlist_t *ilist, std::vector<memref_instr_t> &memref_instr_vec,
+                    addr_t base_addr)
+{
+    static const int MAX_DECODE_SIZE = 1024;
+    byte decode_buf[MAX_DECODE_SIZE];
+    byte *pc =
+        instrlist_encode_to_copy(GLOBAL_DCONTEXT, ilist, decode_buf,
+                                 reinterpret_cast<app_pc>(base_addr), nullptr, true);
+    assert(pc != nullptr);
+    assert(pc <= decode_buf + sizeof(decode_buf));
+    std::vector<memref_t> memrefs = {};
+    memrefs.push_back(
```

Comment:
What if the user already added a TRACE_MARKER_TYPE_FILETYPE marker in memref_instr_vec?